### PR TITLE
add : Best time to buy and sell stock program under GREEDY methods

### LINKED ISSUE #10113: [Add] : Best time to buy and sell stock program by using GREEDY
### Feature description

Best time to buy and sell stock program - Under GREEDY METHODS

image

image

image

### PR DESCRIPTION
**Pull request by kosuri-indu**
### Describe your change:

This PR is to add the Best time to buy and sell stock program under greedy methods. 
Fixes #10113

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by kosuri-indu** [comment]
Hello @cclauss , Could you review this PR ?

**Comment by kosuri-indu** [comment]
Hey @cclauss , Could you review this pull request and suggest any changes ?

File: greedy_methods/best_time_to_buy_and_sell_stock.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return 0
+
+    min_price = prices[0]
+    max_profit: int = 0
+
+    for price in prices:
+        if price < min_price:
+            min_price = price
+        elif price - min_price > max_profit:
+            max_profit = price - min_price
```
Comment (cclauss) [inline greedy_methods/best_time_to_buy_and_sell_stock.py:37]:
```suggestion
        min_price = max(price, min_price)
        max_profit = max(price - min_price, max_profit)
```

Comment (kosuri-indu) [reply greedy_methods/best_time_to_buy_and_sell_stock.py:37]: @cclauss I think it will be 
min_price = min(price, min_price)

**Comment by kosuri-indu** [comment]
@cclauss Thank you for updating :) !

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Buying and selling stocks is when you have all the quotes and then can go back in time and buy and sell.

Like shooting fish in a barrel 🐟

**Comment by kosuri-indu** [comment]
That is true 

> Buying and selling stocks is trivial when you have all the quotes and then can go back in time and buy and sell.
> 
> Like shooting fish in a barrel 🐟

That is true hehe

**Pull request merged**